### PR TITLE
Updated release versions for azure-cosmos 4.17.0, azure-spring-data-cosmos 3.9.0

### PR DESCRIPTION
**Pull request by kushagraThapar**
#### New Features
* Adjust MicroBatchSize dynamically based on throttling rate in BulkExecutor.

#### Key Bug Fixes
* Fixed an issue where authentication context was missing in Barrier related requests.
* Fixed an issue where `ChangeFeedProcessor` was resuming from zero continuation token for new partitions on partition splits.

#### New Features
* Added `PartitionKey` support to spring data single partition queries.
* Updated `Spring Boot` to version 2.5.2.
* Updated `azure-cosmos` to version 4.17.0.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,12 @@
 ## Release History
 
-## 4.17.0-beta.1 (Unreleased)
+### 4.17.0 (2021-07-08)
+#### New Features
+* Adjust MicroBatchSize dynamically based on throttling rate in BulkExecutor.
 
+#### Key Bug Fixes
+* Fixed an issue where authentication context was missing in Barrier related requests.
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:8]: this fix some specific to AAD only.

```suggestion
* Fixed an issue where AAD authentication context was missing in Barrier related requests.
```

**Review by moderakh — APPROVED** [review]
minor suggestion on the changelog other than that LGTM. thanks

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: Updated, thanks @moderakh

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: this fix some specific to AAD only also as this is targeting end users who don't know what barrier request is maybe something like this:

```suggestion
* Fixed an issue with AAD authentication in Strong and BoundedStatelness direct mode.
```

**Review by milismsft — APPROVED** [review]

**Pull request merged**